### PR TITLE
[FE] View available pets

### PR DESCRIPTION
**Pull request by qroll**
The scope of this PR includes: display, pagination and basic filtering; currently using mocked data

For the filters, implemented species, age, gender and breed only
  - Species, age, gender have fixed options
  - Breed is derived from the list of pets
  - AND logic between filters, empty state is shown if no matching pets [UX here is a bit dodgy as user can select species=Cat and breed={dog breed} which will never overlap]

I can work on the remaining filters and search function in another PR as this one is getting pretty huge but I think we should prioritise integration or building other pages first.

### CODE REVIEW

**Comment by stonefruit** [comment]
Regarding a comment Al gave previously, the filters and the cards should show corresponding fields. 

Filter Fields: Species, Age, Gender, Breed
Fields on Card: Name, Gender, Breed, Weight

In my opinion, it is ok to have less filters than fields on the card. However we may want to display the Age and Species on the card.

**Comment by stonefruit** [comment]
"Done" button in each filter seems redundant since the filters are instantly running. "Close" might be a more appropriate button label.

**Comment by stonefruit** [comment]
I think we can hide the search bar since I don't think we are doing it for the MVP if I remember correctly.

**Comment by stonefruit** [comment]
We will need to edit the route in the "Adopt" Button in "/" to link to this page. Currently it shows an alert. Doing so will allow simple demo of the adopter's workflow if the product guys want to use it.

File: frontend/components/adopter/ViewPets/components/AnimalListing.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+const GenderDisplay = {
+	F: "Female",
+	M: "Male",
+};
+
+export function AnimalListing(props: AnimalListingProps) {
+	const {
+		animal: { name, gender, breed, weightKg, animalImages },
+	} = props;
+	const thumbnail = animalImages?.[0].thumbnailUrl;
```
Comment (stonefruit) [inline frontend/components/adopter/ViewPets/components/AnimalListing.tsx:18]: Later on, we might want to use a placeholder image to say no available image. Currently it looks like this which should be fine.
Screenshot 2021-12-28 at 11 15 11 PM

File: frontend/components/adopter/ViewPets/data/filters.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            },
+            {
+                label: "10 years and above",
+                value: AgeFilter.Above120Months,
+            },
+        ],
+    },
+];
+
+export const AgeFilterMatcher = new Map<
```
Comment (stonefruit) [inline frontend/components/adopter/ViewPets/data/filters.ts:74]: I played around with this and noticed this method does not enforce all the AgeFilter values to be a key. An alternative would be something like this:

```
export const _AgeFilterMatcher: Record<AgeFilter, (ageMonths: number) => boolean> ={
    [AgeFilter.Above0Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
    [AgeFilter.Above6Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
    [AgeFilter.Above12Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
    [AgeFilter.Above36Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
    [AgeFilter.Above72Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
    [AgeFilter.Above120Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
}
```

File: frontend/components/adopter/ViewPets/data/filters.ts [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    {
+        title: "Beyond a year",
+        options: [
+            {
+                label: "1-3 years",
+                value: AgeFilter.Above12Months,
+            },
+            {
+                label: "4-6 years",
+                value: AgeFilter.Above36Months,
```
Comment (stonefruit) [inline frontend/components/adopter/ViewPets/data/filters.ts:60]: Might be good to rename and/or add unit tests on these filters as the labels do not match the enum name `Above36Months` and the filter does not match the label too. The current value would include 3+ years old which is inaccurate to 4-6 years.

File: frontend/components/adopter/ViewPets/data/filters.ts [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+            },
+            {
+                label: "Male",
+                value: GenderFilter.M,
+            },
+        ],
+    },
+];
+
+function toTitlecase(str: string) {
```
Comment (stonefruit) [inline frontend/components/adopter/ViewPets/data/filters.ts:115]:
```suggestion
function toTitleCase(str: string) {
```

**Comment by stonefruit** [comment]
> 

@qroll I think we should check if this is necessary. For now PR looks good. We can edit to show the data later if needed.

**Review by stonefruit — APPROVED** [review]

**Comment by qroll** [comment]
> > Regarding a comment Al gave previously, the filters and the cards should show corresponding fields.
> > Filter Fields: Species, Age, Gender, Breed Fields on Card: Name, Gender, Breed, Weight
> > In my opinion, it is ok to have less filters than fields on the card.
> > However we may want to display the **Age** and **Species** on the card.
> 
> @qroll I think we should check if this is necessary. For now PR looks good. We can edit to show the data later if needed.

Ok cool, thanks for merging. Was playing around with this, actually, and noticed that the text will start to obscure the picture when more fields are added. If we're enhancing this in the future, some tweaking might be needed for different breakpoints.

**Comment by stonefruit** [comment]
Okie noted

**Pull request merged**